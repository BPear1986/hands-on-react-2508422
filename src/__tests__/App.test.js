import React from "react"
import renderer from "react-test-renderer"
import App from"../App"

beforeEach(() => {
  fetch.resetMocks();
});

it("test runs successfuly", ()=>{
  const truth = true
  expect(truth).toEqual(true)
})

it("App snapshot test", async () => {
  fetch.mockResponseOnce(JSON.stringify([{
    memberInfo: [{
    id: 0,
    name: "Aneirin",
    slug: "aneirin",
    bio: "Aneirin is the foremost expert of cold climates in planetoids universe-wide, with a particular inquisitiveness in areas colder than -300 F / - 184.4444444 C (hence the profound personal interest in knitting mittens). Having travelled extensively, Aneirin enjoys collecting, receiving, and sending distinctive postcards from all corners of the macrocosm.",
    origin: "Zacon II",
    eyes: 2,
    hobby: "knitting",
    ambulation: "walking, skipping",
    job_title: "Cold Climate Analyst",
    dream_job_title: "Private Island Caretaker",
    singing_voice: "alto",
    favorites: {
      food: "asparagus",
      music_genre: "hip hop, classic rock",
      book: "The Call of the Wild by Jack London",
      footwear: "cowboy boots"
    }
  },{
    id: 1,
    name: "Celestia Dristi",
    slug: "celestia-dristi",
    bio: "As a mobility engineer, Celestia Dristi helps individuals and companies to improve personal and  professional agility and growth in technology and systems. An expert in translating and implementing  needs into functionality, Celestia Dristi has a unique proclivity for solving problems and helping to   avoid future issues.",
    origin: "Krachonk",
    eyes: 1,
    hobby: "landscaping",
    ambulation: "rolling",
    job_title: "Mobility Engineer",
    dream_job_title: "Mime",
    singing_voice: "soprano",
    favorites: {
      food: "ice cream cake",
      music_genre: "reggae",
      book: "A Tale of Two Cities by Charles Dickens",
      footwear: "n/a"
    }}]}]));

  const component = renderer.create(<App/>);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot()
})
