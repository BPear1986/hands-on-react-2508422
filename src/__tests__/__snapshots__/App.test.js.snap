// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot test 1`] = `
[
  <div
    className="navContainer"
  >
    <nav
      className="container"
    >
      <ul>
        <li>
          <a
            href="#"
          >
            <img
              alt="Stargazers logo"
              src="images/logo_bug_stargazers.svg"
              style={
                {
                  "height": "80px",
                }
              }
            />
          </a>
        </li>
      </ul>
      <ul>
        <li>
          <details
            role="list"
          >
            <summary
              aria-haspopup="listbox"
              role="link"
            >
              cast
            </summary>
            <ul
              role="listbox"
            />
          </details>
        </li>
      </ul>
    </nav>
  </div>,
  <div
    className="container"
  >
    <hgroup>
      <img
        alt="StarGazers Group"
        src="images/group.svg"
      />
      <h1>
        Meet the StarGazers
      </h1>
      <p>
        Members of an 
        <b>
          intergalactic alliance
        </b>
         paving the way for peace and benevolence among all species. They are known for their enthusiasm for science, for their love of fun, and their dedication to education.
      </p>
      <div
        style={
          {
            "display": "grid",
            "gap": "1rem",
            "gridTemplateColumns": "repeat(auto-fit, minmax(90px, 1fr))",
            "marginBottom": "1rem",
          }
        }
      />
    </hgroup>
  </div>,
]
`;
